src/cmap/auth/mongodb_oidc/callback_workflow.ts

```diff
@@ -71,12 +70,12 @@ export class CallbackWorkflow implements Workflow {
         } catch (error) {
           // If authentication errors when using a cached token we remove it from
           // the cache.
-          this.cache.deleteEntry(connection.address, credentials.username);
-          return Promise.reject(error);
+          this.cache.deleteEntry(connection.address, credentials.username || '');
```

Comment:
Yes - since we moved username to the second parameter of the method it could no longer be optional in the signature without some ugly overloads and type checking, which I find just messy. So we just ensure here it's always set to something.